TwitchPlaysAssembly/Src/UI/ModuleCameras.cs

```diff
@@ -344,54 +261,43 @@ public IEnumerator UnzoomCamera(BombComponent component, float delay)
 		return null;
 	}
 
-	public bool TryViewModule(TwitchComponentHandle handle)
+	public bool TryViewModule(TwitchComponentHandle module)
 	{
-		if (handle == null)
+		if (module == null)
 			return false;
 
-		var item = EnsureModuleItem(handle);
-		handle.LastUsed = DateTime.UtcNow;
+		module.LastUsed = DateTime.UtcNow;
 
 		// Is the module already viewed?
-		int existingCamera = CurrentModulesContains(handle.bombComponent);
+		int existingCamera = CurrentModulesContains(module.bombComponent);
 		if (existingCamera > -1)
 			return false;
 
 		// Find a camera
-		var camera = AvailableCamera(handle.CameraPriority);
+		var camera = AvailableCamera(module.CameraPriority);
 		if (camera == null)
 			return false;
 
 		// If we can replace a LOWER-priority camera (or an unused slot), do so before enabling the camera wall.
-		if (camera.Module == null || camera.Module.Handle.CameraPriority < handle.CameraPriority)
-			camera.ViewModule(item);
+		if (camera.Module == null || camera.Module.CameraPriority < module.CameraPriority)
+			camera.ViewModule(module);
 		// If we can and should enable the camera wall, enable it and then view the module.
-		else if (AutomaticCameraWallEnabled && !CameraWallEnabled && _cameras.Where(cam => cam.Module != null && cam.Module.Handle.Claimed).Select(cam => cam.Module.Handle.PlayerName).Distinct().Count() >= 6)
+		else if (AutomaticCameraWallEnabled && !CameraWallEnabled && _cameras.Where(cam => cam.Module != null && cam.Module.Claimed).Select(cam => cam.Module.PlayerName).Distinct().Count() >= 6)
 		{
 			EnableCameraWall();
-			TryViewModule(handle);
+			TryViewModule(module);
 		}
 		// If the camera is already enabled, replace a suitable SAME-priority camera.
 		else
-			camera.ViewModule(item);
+			camera.ViewModule(module);
 		return true;
 	}
 
-	public ModuleItem EnsureModuleItem(TwitchComponentHandle handle)
-	{
-		if (!_moduleItems.TryGetValue(handle.bombComponent, out var item))
-		{
-			item = new ModuleItem(handle.bombComponent, handle);
-			_moduleItems.Add(handle.bombComponent, item);
-		}
-		return item;
-	}
-
 	private ModuleCamera AvailableCamera(CameraPriority maxPriority) => _cameras
-				.Where(c => c.Module == null || (c.Module.Handle != null && c.Module.Handle.CameraPriority <= maxPriority))
+				.Where(c => c.Module == null || (c.Module != null && c.Module.CameraPriority <= maxPriority))
 				.OrderBy(c => c.Module != null)
-				.ThenByDescending(c => c.Module?.Handle.CameraPriority)
-				.ThenBy(c => c.Module?.Handle.LastUsed)
+				.ThenByDescending(c => c.Module?.CameraPriority)
+				.ThenBy(c => c.Module?.LastUsed)
```

Comment:
Module is a MonoBehaviour and so we should do a explicit null check for these two lines.